chalice/app.py

```diff
@@ -0,0 +1,80 @@
+import multiprocessing
+import traceback
+
+from botocore.exceptions import ClientError
+from chalice import Chalice, BadRequestError, NotFoundError
+from chalicelib.matrix_handler import LoomMatrixHandler
+from chalicelib.request_handler import RequestHandler, RequestStatus
+
+app = Chalice(app_name='matrix-service')
+
+mtx_handler = LoomMatrixHandler()
+
+
+@app.route('/matrices/health')
+def health():
+    return {'status': 'OK'}
+
+
+@app.route('/matrices/concat/{request_id}', methods=['GET'])
+def check_request_status(request_id):
+    """
+    Check the status of a matrices concatenation request based on
+    a request ID.
+
+    :param request_id: <string> Matrices concatenation request ID
+    """
+    try:
+        app.log.info("Checking request({}) status.".format(request_id))
+        request_status = RequestHandler.check_request_status(request_id)
+        app.log.info("Request({}) status: {}.".format(request_id, request_status))
+
+        if request_status == RequestStatus.UNINITIALIZED:
+            raise NotFoundError("Request({}) does not exist.".format(request_id))
```

Comment:
Because of AWS's consistency model, you will likely run into scenarios where the request status will report uninitialized even though it's not really the case.  You could solve the problem by never GET'ing a file before you PUT it, but it adds a bit of complexity to your design.

https://docs.aws.amazon.com/AmazonS3/latest/dev/Introduction.html#ConsistencyModel